### PR TITLE
[RW-2708][Risk=no] Srubenst/notebook locking ux

### PR DESCRIPTION
**Pull request by blrubenstein**
This does not tackle the jupyter extension around locking. That I am going to do in a separate PR. Also fixes a key mismatch in the API with the gcs metadata.

### CODE REVIEW

File: ui/src/app/views/notebook-in-use-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      </ModalBody>
+      <ModalFooter style={{alignItems: 'center'}}>
+        <Button type='secondary'
+                style={{width: '6rem', margin: '0 10px'}}
+                onClick={() => { this.props.onCancel(); }}>
+          Cancel
+        </Button>
+        <Button type='secondary'
+                style={{width: '8rem', margin: '0 10px'}}
+                onClick={() => { this.setState({copyLoading: true}); this.props.onCopy(); }}>
```
Comment (calbach) [inline ui/src/app/views/notebook-in-use-modal.tsx]: nit: newlines?

File: ui/src/app/views/interactive-notebook.tsx [hunk 2]
```diff
@@ -168,15 +191,37 @@ export const InteractiveNotebook = fp.flow(withUrlParams(), withCurrentWorkspace
       return WorkspacePermissionsUtil.canWrite(this.props.workspace.accessLevel);
     }
 
+    private cloneNotebook() {
+      const {ns, wsid, nbName} = this.props.urlParams;
+      workspacesApi().cloneNotebook(
+        ns, wsid, nbName).then((notebook) => {
```
Comment (calbach) [inline ui/src/app/views/interactive-notebook.tsx]: nit: params can go on previous line

File: ui/src/app/views/interactive-notebook.tsx [hunk 3]
```diff
@@ -168,15 +191,37 @@ export const InteractiveNotebook = fp.flow(withUrlParams(), withCurrentWorkspace
...
+          navigateByUrl(
+            `/workspaces/${ns}/${wsid}/notebooks/${encodeURIComponent(notebook.name)}`);
+        });
+    }
+
+    private get notebookInUse() {
+      const {lastLockedBy, lockExpirationTime} = this.state;
+      return lastLockedBy !== null && lockExpirationTime - new Date().getTime() >= 0
```
Comment (calbach) [inline ui/src/app/views/interactive-notebook.tsx]:
```suggestion
      return lastLockedBy !== null && new Date().getTime()  < lockExpirationTime
```

**Comment by blrubenstein** [comment]
Screen Shot 2019-07-24 at 9 59 56 AM

**Pull request merged**